File: alica_engine/src/engine/modelmanagement/factories/TaskRepositoryFactory.cpp
#include "engine/modelmanagement/factories/TaskRepositoryFactory.h"

#include "engine/model/Task.h"
#include "engine/model/TaskRepository.h"
#include "engine/modelmanagement/Strings.h"
#include "engine/modelmanagement/factories/TaskFactory.h"

namespace alica
{
TaskRepository* TaskRepositoryFactory::create(const YAML::Node& node)
{
    auto* taskRepository = new TaskRepository();
    Factory::setAttributes(node, taskRepository);
    Factory::storeElement(taskRepository, alica::Strings::taskrepository);

    if (Factory::isValid(node[alica::Strings::tasks])) {
        const YAML::Node& tasksNode = node[alica::Strings::tasks];
        for (YAML::const_iterator it = tasksNode.begin(); it != tasksNode.end(); ++it) {
            taskRepository->addTask(TaskFactory::create(*it, taskRepository));
        }
    }

    return taskRepository;
}
} // namespace alica

File: supplementary/plan-conversion-tool/src/main/java/com/rapyutarobotics/alica/factories/TaskRepositoryFactory.java
package com.rapyutarobotics.alica.factories;

import com.rapyutarobotics.alica.ConversionProcess;
import com.rapyutarobotics.alica.Tags;
import de.unikassel.vs.alica.planDesigner.alicamodel.Task;
import de.unikassel.vs.alica.planDesigner.alicamodel.TaskRepository;
import org.w3c.dom.Element;
import org.w3c.dom.NodeList;

public class TaskRepositoryFactory extends Factory {

    public static TaskRepository create(Element taskRepositoryNode, ConversionProcess cp) {
        TaskRepository taskRepository = new TaskRepository();
        Factory.setAttributes(taskRepositoryNode, taskRepository);
        cp.addElement(taskRepository);

        NodeList tasksList = taskRepositoryNode.getElementsByTagName(Tags.TASKS);
        for (int i = 0; i < tasksList.getLength(); i++) {
            Element taskNode = (Element) tasksList.item(i);
            Task task = TaskFactory.create(taskNode, cp);
            taskRepository.addTask(task);
            task.setTaskRepository(taskRepository);
        }

        return taskRepository;
    }
}